service/resource/s3objectv2/current.go

```diff
@@ -11,49 +12,67 @@ import (
 )
 
 func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
-	output := BucketObjectState{}
+	output := map[string]BucketObjectState{}
 	customObject, err := keyv2.ToCustomObject(obj)
 	if err != nil {
 		return output, microerror.Mask(err)
 	}
 
 	r.logger.LogCtx(ctx, "debug", "looking for S3 objects")
 
-	workersObjectName := keyv2.BucketObjectName(customObject, prefixWorker)
-
 	accountID, err := r.awsService.GetAccountID()
 	if err != nil {
 		return output, microerror.Mask(err)
 	}
 
 	bucketName := keyv2.BucketName(customObject, accountID)
-
-	input := &s3.GetObjectInput{
-		Key:    aws.String(workersObjectName),
+	input := &s3.ListObjectsV2Input{
 		Bucket: aws.String(bucketName),
 	}
-	result, err := r.awsClients.S3.GetObject(input)
+	result, err := r.awsClients.S3.ListObjectsV2(input)
+	if err != nil {
+		return output, microerror.Mask(err)
+	}
 
-	if IsObjectNotFound(err) || IsBucketNotFound(err) {
-		r.logger.LogCtx(ctx, "debug", "did not find the S3 objects")
-		// fall through
+	r.logger.LogCtx(ctx, "debug", "found the S3 objects")
 
-	} else if err != nil {
+	for _, object := range result.Contents {
+		body, err := r.getBucketObjectBody(bucketName, *object.Key)
+		if err != nil {
+			return output, microerror.Mask(err)
+		}
 
-		return output, microerror.Mask(err)
+		output[*object.Key] = BucketObjectState{
+			Body:   body,
+			Bucket: bucketName,
+			Key:    *object.Key,
+		}
+	}
 
-	} else {
+	return output, nil
+}
 
-		r.logger.LogCtx(ctx, "debug", "found the S3 objects")
-		output.WorkerCloudConfig.Key = workersObjectName
-		output.WorkerCloudConfig.Bucket = bucketName
+func (r *Resource) getBucketObjectBody(bucketName string, keyName string) (string, error) {
+	var body string
 
-		buf := new(bytes.Buffer)
-		buf.ReadFrom(result.Body)
-		body := buf.String()
+	input := &s3.GetObjectInput{
+		Bucket: aws.String(bucketName),
+		Key:    aws.String(keyName),
+	}
+	result, err := r.awsClients.S3.GetObject(input)
+	if IsObjectNotFound(err) || IsBucketNotFound(err) {
+		r.logger.Log("debug", fmt.Sprintf("did not find S3 object '%s'", keyName))
```

Comment:
Not sure but this is maybe worth a warning level.

Reply:
Good point. I've increased the log level to info. As this shoudn't happen. If it does we will repeat the PutObject API call which I think is the correct behavior.